pkg/store/bucket.go

```diff
@@ -709,6 +715,7 @@ func (s *BucketStore) addBlock(ctx context.Context, meta *metadata.Meta) (err er
 			level.Warn(s.logger).Log("msg", "loading block failed", "elapsed", time.Since(start), "id", meta.ULID, "err", err)
 		} else {
 			level.Info(s.logger).Log("msg", "loaded new block", "elapsed", time.Since(start), "id", meta.ULID)
+			s.metrics.blockLoadDuration.Observe(time.Since(start).Seconds())
```

Comment:
Do we want to record the time of mainly `NewBinaryReader` function?
This will include cases where we hit the local cache and not load index header from object storage.

Do we have a better way to track the block load duration? But I am fine to leave it as it is. P99 can usually tell the case where we need to download remotely.

Reply:
I could add another metric for specifically measuring the time taken for `WriteBinary()`.